src/data/json-update-builder.js

```diff
@@ -0,0 +1,79 @@
+const OPERATIONS = {
+  SET         : 'JSON_SET',
+  INSERT      : 'JSON_INSERT',
+  REPLACE     : 'JSON_REPLACE',
+  REMOVE      : 'JSON_REMOVE',
+  ARRAY_APPEND: 'JSON_ARRAY_APPEND',
+  ARRAY_INSERT: 'JSON_ARRAY_INSERT'
+}
+
+const OPERATIONS_JOINED = Object.keys(OPERATIONS).join('\', \'')
+
+const OPERATION_FIELD_NAME = '___operation'
+const ARGS_FIELD_NAME = 'args'
+
+export default class JSONUpdateBuilder {
+  constructor(operationName, args) {
```

Comment:
to avoid the passing `args` I propose to create separated classes for each operation

```
class JSONSetBuilder extends JSONUpdateBuilder
class JSONInsertBuilder extends JSONUpdateBuilder
class JSONReplaceBuilder extends JSONUpdateBuilder
class JSONRemoveBuilder extends JSONUpdateBuilder
class JSONArrayAppendBuilder extends JSONUpdateBuilder
class JSONArrayInsertBuilder extends JSONUpdateBuilder
``` 

where each class has its own `args` type and `addArgument ` method

or we can have a list of specific classes, for instance: JSONRemoveBuilder
where will be overridden this `args` in the constructor and `addArgument ` method
in that case, the JSONRemoveBuilder class will always work with `args` as object

@stanislav-grin what do you think?

Reply:
I don't like your idea. How about this:

```
class JSONSetBuilder extends JSONBasicUpdateBuilder
...
class JSONRemoveBuilder extends JSONBasicUpdateBuilder
...
```

and in the class JSONUpdateBuilder
```
static SET() {
    return new JSONSetBuilder()
  }
...
static REMOVE() {
    return new JSONRemoveBuilder()
  }
...

```

Reply:
or 
static SET() {
    return new JSONBasicUpdateBuilder()
  }
...
static REMOVE() {
    return new JSONRemoveBuilder()
  }
...

Reply:
@Igor-aka-Gillirond 
you described completely the same what wrote above =)

Reply:
JSONUpdateBuilder 
->
 return new JSONRemoveBuilder in REMOVE METHOD 
->
 JSONBasicUpdateBuilder